tests/console/check_migration_features.pm

```diff
@@ -40,20 +40,25 @@ sub check_python2_dropped {
 # SLE-23610: Python3 module
 # test steps:
 # 1) activate the python3 module
-# 2) install the python310 package
-# 3) check python3.10's version which should be 3.10.X
+# 2) install the python31X package
+# 3) check python3.XX's version which should be 3.XX.X
 # 4) check python3's version
-# 5) check python310's lifecycle
+# 5) check python3XX's lifecycle
 sub check_python3_module {
     record_info('SLE-23610', 'Check Python3 Module');
     my $OS_VERSION = script_output("grep VERSION_ID /etc/os-release | cut -c13- | head -c -2");
     my $ARCH = get_required_var('ARCH');
     assert_script_run("SUSEConnect -p sle-module-python3/$OS_VERSION/$ARCH", timeout => 180);
-    zypper_call("se python310");
-    zypper_call("in python310");
-    assert_script_run("python3.10 --version | grep Python | grep 3.10.");
+    my $python_version = get_var('PYTHON_VERSIONS', '3.11');
+    my $sub_version = substr($python_version, 2);
+    my @python31x_results = split "\n", script_output("zypper se python3$sub_version | awk -F \'|\' \'/python3$sub_version/ {gsub(\" \", \"\"); print \$2}\'");
+    if ((scalar(@python31x_results) == 0) || ($python31x_results[0] ne "python3$sub_version")) {
+        die("Python $python_version not found in the repositories\n\n");
+    }
```

Comment:
Why not just zypper_call("se python3$sub_version"); ? 
We can have a variable to replace  the " python3$sub_version", such as my $python_pkg = "python3"$sub_version;